src/common/bzip2_op/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 add_library(bzip2_op STATIC src/bzip2_op.c)
 
+get_filename_component(COMMON_FOLDER ${CMAKE_CURRENT_SOURCE_DIR}/../ ABSOLUTE)
 target_include_directories(bzip2_op PUBLIC include)
+include_directories(${COMMON_FOLDER}/pal/include)
+include_directories(${COMMON_FOLDER}/error_messages/include)
```

Comment:
Is it better to include the directories like this than to link against the targets that normally bring them? I mean, instead of adding the directories just using target_link_libraries with pal and error_messages.

Reply:
error_messages are almost always included for definitions, and pal also solves incompatibilities in the header but not in any linkable file.

Reply:
Since `error_messages` is an `INTERFACE` target, it could be linked with pretty much the same result. Again, not wrong but manually picking up folders feels less encapsulated than using a target.

Maybe this is not the case, but imagein if a target comes to declare an entity in the include folder and provide its definition in a source file, then we would have a linker error because instead of linking against the target, we just included the `include` folder.

Reply:
Done